Lib/asyncio/queues.py

```diff
@@ -121,6 +121,10 @@ def full(self):
                 await putter
             except:
                 putter.cancel()  # Just in case putter is not done yet.
+                try:
+                    self._putters.remove(putter)
+                except ValueError:
+                    pass
```

Comment:
So I applied this patch and changed `pass` to `raise`.  Nothing happened, `test_cancelled_putters_not_being_held_in_self_putters` passed just fine.  Putting a `print()` before `raise` showed that this `except` block isn't triggered by the test at all.

All in all, I don't understand this fix.  Please make sure that the test (a) triggers the code you change; (b) there's a clear comment in the code explaining why/what it's doing.

Reply:
Added a refactor of the test with the proper comments, comments to the bugfix explaining it, and added also another test to ensure that the except block is correctly executed. 

The requested changes are really similar to the merged code of the following link, it is the same memory leak but in the getters instead of the putters.

https://github.com/python/cpython/commit/c62f0cb3b1f6f9ca4ce463b1c99b0543bdfa38d6#diff-3947468b604f664993c24d2b5560635c

I think that now the code is much more clear and easy to understand, thx for the correction.